### PR TITLE
Simplify targeted-refresh subclasses

### PR DESCRIPTION
**Pull request by agrare**
Reduce the number of methods that providers have to override from core inventory classes

Depends on:
- [x] https://github.com/ManageIQ/manageiq/pull/21225

### CODE REVIEW

File: spec/models/manageiq/providers/oracle_cloud/cloud_manager/refresher_spec.rb [hunk 1]
```diff
@@ -184,7 +41,7 @@ def assert_specific_cloud_volume
         with_vcr("vm_target") { refresh(ems_event.manager_refresh_targets) }
 
         vm.reload
-        expect(vm.raw_power_state).to eq("STOPPED")
+        assert_specific_instance(:expected_raw_power_state => "STOPPED", :expected_power_state => "off")
```
Comment (agrare) [inline spec/models/manageiq/providers/oracle_cloud/cloud_manager/refresher_spec.rb:44]: The existing targeted refresh spec didn't show the failures of having the wrong strategy, and this is a better test anyway

File: app/models/manageiq/providers/oracle_cloud/cloud_manager/vm.rb [hunk 2]
```diff
@@ -46,7 +46,7 @@ def self.calculate_power_state(raw_power_state)
     case raw_power_state.downcase
     when /running/, /starting/
       "on"
-    when /shutdown/, /stopping/, /terminating/, /terminated/
+    when /shutdown/, /stopping/, /stopped/, /terminating/, /terminated/
```
Comment (agrare) [inline app/models/manageiq/providers/oracle_cloud/cloud_manager/vm.rb:49]: Found an actual bug :tada:

**Comment by agrare** [comment]
@Fryguy this is green

**Comment by Fryguy** [comment]
Backported to `morphy` in commit fea9057615a1d93b3053d6241481366891160387.

```text
commit fea9057615a1d93b3053d6241481366891160387
Author: Jason Frey <fryguy9@gmail.com>
Date:   Wed May 19 16:52:14 2021 -0400

    Merge pull request #24 from agrare/simplify_provider_inventory_overrides
    
    Simplify targeted-refresh subclasses
    
    (cherry picked from commit 56cea8f7abd8443602f8958c967c80054b54ee21)
```

**Pull request merged**